website/static/js/fileBrowser.js

```diff
@@ -0,0 +1,799 @@
+                    'data-placement' : 'bottom'
+                }, ''),
+                m('.pull-right', m('button.btn.btn-xs.btn-success[data-toggle="modal"][data-target="#addColl"]', m('i.fa.fa-plus')))
+            ]),
+            m('ul', [
+                args.collections.map(function(item, index){
+                    if (item.id === args.activeFilter()) {
+                        selectedCSS = 'active';
+                    } else if (item.id === args.collectionMenuObject().item.id) {
+                        selectedCSS = 'bg-color-hover';
+                    } else {
+                        selectedCSS = '';
+                    }
+                    if (!item.data.systemCollection) {
+                        submenuTemplate = m('i.fa.fa-ellipsis-v.pull-right.text-muted.p-xs', {
+                            onclick : function (e) {
+                                args.updateCollectionMenu(item, e);
+                            }
+                        });
+                    } else {
+                        submenuTemplate = '';
+                    }
+                    return m('li', { className : selectedCSS, 'data-index' : index },
+                        [
+                            m('a', { href : '#', onclick : args.updateFilter.bind(null, item) },  item.label),
+                            submenuTemplate
+                        ]
+                    );
+                }),
+                args.showCollectionMenu() ? m('.collectionMenu',{
+                    style : 'position:absolute;top: ' + args.collectionMenuObject().y + 'px;left: ' + args.collectionMenuObject().x + 'px;'
+                }, [
+                    m('.menuClose', { onclick : function (e) {
+                            args.showCollectionMenu(false);
+                            args.resetCollectionMenu();
+                        }
+                    }, m('.text-muted','×')),
+                    m('ul', [
+                        m('li[data-toggle="modal"][data-target="#renameColl"].pointer',{
+                            onclick : function (e) {
+                                args.showCollectionMenu(false);
+                            }
+                        }, [
+                            m('i.fa.fa-pencil'),
+                            ' Rename'
+                        ]),
+                        m('li[data-toggle="modal"][data-target="#removeColl"].pointer',{
+                            onclick : function (e) {
+                                args.showCollectionMenu(false);
+                            }
+                        }, [
+                            m('i.fa.fa-trash'),
+                            ' Delete'
+                        ])
+                    ])
+                ]) : ''
+            ]),
+            m('.fb-collections-modals', [
+                m('#addColl.modal.fade[tabindex=-1][role="dialog"][aria-labelledby="addCollLabel"][aria-hidden="true"]',
+                    m('.modal-dialog',
```

Comment:
There's a lot of repetition in the view for this component. Can you factor some of the common code into smaller, stateless components? These modal dialogs seem like a good candidate.